packages/graphql/tests/e2e/subscriptions/single-instance-plugin.e2e.test.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Driver } from "neo4j-driver";
+import supertest, { Response } from "supertest";
+import { Neo4jGraphQL } from "../../../src/classes";
+import { generateUniqueType } from "../../utils/graphql-types";
+import { ApolloTestServer, TestGraphQLServer } from "../setup/apollo-server";
+import { WebSocketTestClient } from "../setup/ws-client";
+import neo4j from "../setup/neo4j";
+import { Neo4jGraphQLSubscriptionsSingleInstancePlugin } from "../../../src";
+import { delay } from "../../../src/utils/utils";
+
+describe("Create Subscription", () => {
+    let driver: Driver;
+    let plugin: Neo4jGraphQLSubscriptionsSingleInstancePlugin;
+
+    const typeMovie = generateUniqueType("Movie");
+
+    const subscriptionQuery = `subscription {
+                            ${typeMovie.operations.subscribe.created} {
+                                ${typeMovie.operations.subscribe.payload.created} {
+                                    title
+                                }
+                                event
+                                timestamp
+                            }
+                        }`;
+
+    let server: TestGraphQLServer;
+
+    let wsClient: WebSocketTestClient;
+    let wsClient2: WebSocketTestClient;
+
+    beforeAll(async () => {
+        plugin = new Neo4jGraphQLSubscriptionsSingleInstancePlugin();
+        const typeDefs = `
+         type ${typeMovie} {
+             title: String
+         }
+         `;
+
+        driver = await neo4j();
+
+        const neoSchema = new Neo4jGraphQL({
+            typeDefs,
+            driver,
+            plugins: {
+                subscriptions: plugin,
+            },
+        });
+
+        server = new ApolloTestServer(neoSchema);
+        await server.start();
+    });
+
+    beforeEach(() => {
+        wsClient = new WebSocketTestClient(server.wsPath);
+        wsClient2 = new WebSocketTestClient(server.wsPath);
+    });
+
+    afterEach(async () => {
+        await wsClient.close();
+        await wsClient2.close();
+    });
+
+    afterAll(async () => {
+        await server.close();
+        await driver.close();
+    });
+
+    // NOTE: This test **may** be flaky, if so, feel free to remove it
+    test("listeners are properly triggered and cleaned up", async () => {
```

Comment:
question: Would adding a delay of 100ms before checking the listenerCount make the test less flaky?

Reply:
The flakiness should be fixed by the 50ms delay that is already in place (the comment refers to those). In my tests, I didn't see any problem in the first check